File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/levels/traps/PoisonTrap.java

package com.watabou.pixeldungeon.levels.traps;

import com.nyrds.pixeldungeon.levels.objects.ITrigger;
import com.watabou.pixeldungeon.Dungeon;
import com.watabou.pixeldungeon.actors.Actor;
import com.watabou.pixeldungeon.actors.Char;
import com.watabou.pixeldungeon.actors.buffs.Buff;
import com.watabou.pixeldungeon.actors.buffs.Poison;
import com.watabou.pixeldungeon.effects.CellEmitter;
import com.watabou.pixeldungeon.effects.particles.PoisonParticle;

import org.jetbrains.annotations.Nullable;

public class PoisonTrap implements ITrigger {

	// 0xBB66EE
	
	public static void trigger( int pos, @Nullable Char ch ) {
		if (ch == null){
			ch = Actor.findChar(pos);
		}
		if (ch != null) {
			Buff.affect( ch, Poison.class,Poison.durationFactor( ch ) * (4 + Dungeon.depth / 2) );
		}
		CellEmitter.center( pos ).burst( PoisonParticle.SPLASH, 3 );
	}

	@Override
	public void doTrigger(int cell, Char ch) {
		trigger(cell,ch);
	}
}

File: scripts/traps/PoisonTrap.lua
local RPD = require "scripts/lib/commonClasses"
local trap = require "scripts/lib/trap"

-- java PoisonTrap: poison the character on the cell (falls back to whoever
-- stands there when the trap was set off by an item), splash visual either way.
return trap.init(
    function (cell, char, data)
        if char == nil then
            char = RPD.Actor:findChar(cell)
        end
        if char ~= nil then
            local duration = RPD.CharUtils:durationFactor(char) * (4 + math.floor(RPD.Dungeon.depth / 2))
            RPD.Buffs.Buff:affect(char, "Poison", duration)
        end
        RPD.Sfx.CellEmitter:center(cell):burst(RPD.Sfx.PoisonParticle.SPLASH, 3)
    end
)
